homeassistant/components/smlight/switch.py

```diff
@@ -102,7 +102,10 @@ async def set_smlight(self, state: bool) -> None:
     @callback
     def event_callback(self, event: SettingsEvent) -> None:
         """Handle switch events from the SLZB device."""
-        if event.setting is not None:
+        if (
+            isinstance(self.coordinator, SmDataUpdateCoordinator)
+            and event.setting is not None
+        ):
```

Comment:
you should overwrite the coordinator which is used, just like how the entity_description type is overwritten

Reply:
oh yes, of course..